sunpy/coordinates/tests/test_frames.py

```diff
@@ -224,12 +224,13 @@ def test_hpc_default_observer():
 
 
 def test_hpc_low_precision_float_warning():
-    hpc = Helioprojective(u.Quantity(0, u.deg, dtype=np.float32),
-                          u.Quantity(0, u.arcsec, dtype=np.float16),
-                          observer=HeliographicStonyhurst(0*u.deg, 0*u.deg, 1*u.AU))
+    with np.errstate(over='ignore'):
```

Comment:
What's this for ooi?

Reply:
For this:

```
=================================== FAILURES ===================================
_____________________ test_hpc_low_precision_float_warning _____________________
[gw0] linux -- Python 3.12.2 /home/runner/work/sunpy/sunpy/.tox/py312-devdeps/bin/python

    def test_hpc_low_precision_float_warning():
>       hpc = Helioprojective(u.Quantity(0, u.deg, dtype=np.float32),
                              u.Quantity(0, u.arcsec, dtype=np.float16),
                              observer=HeliographicStonyhurst(0*u.deg, 0*u.deg, 1*u.AU))

../../.tox/py312-devdeps/lib/python3.12/site-packages/sunpy/coordinates/tests/test_frames.py:227: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../.tox/py312-devdeps/lib/python3.12/site-packages/sunpy/coordinates/frames.py:151: in __init__
    super().__init__(*args, **kwargs)
../../.tox/py312-devdeps/lib/python3.12/site-packages/astropy/coordinates/baseframe.py:317: in __init__
    data = self._infer_data(args, copy, kwargs)  # possibly None.
../../.tox/py312-devdeps/lib/python3.12/site-packages/astropy/coordinates/baseframe.py:470: in _infer_data
    representation_data = representation_cls(copy=copy, **repr_kwargs)
../../.tox/py312-devdeps/lib/python3.12/site-packages/astropy/coordinates/representation/spherical.py:55: in __init__
    super().__init__(lon, lat, differentials=differentials, copy=copy)
../../.tox/py312-devdeps/lib/python3.12/site-packages/astropy/coordinates/representation/base.py:643: in __init__
    super().__init__(*args, **kwargs)
../../.tox/py312-devdeps/lib/python3.12/site-packages/astropy/coordinates/representation/base.py:205: in __init__
    self.attr_classes[component](attr, copy=copy, subok=True)
../../.tox/py312-devdeps/lib/python3.12/site-packages/astropy/coordinates/angles/core.py:604: in __new__
    self._validate_angles()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Latitude 0. arcsec>, angles = <Latitude 0. arcsec>

    def _validate_angles(self, angles=None):
        """Check that angles are between -90 and 90 degrees.
        If not given, the check is done on the object itself.
        """
        # Convert the lower and upper bounds to the "native" unit of
        # this angle.  This limits multiplication to two values,
        # rather than the N values in `self.value`.  Also, the
        # comparison is performed on raw arrays, rather than Quantity
        # objects, for speed.
        if angles is None:
            angles = self
    
        # For speed, compare using "is", which is not strictly guaranteed to hold,
        # but if it doesn't we'll just convert correctly in the 'else' clause.
        if angles.unit is u.deg:
            limit = 90
        elif angles.unit is u.rad:
            limit = 0.5 * np.pi
        else:
            limit = u.degree.to(angles.unit, 90.0)
    
        angles_view = angles.view(np.ndarray)
        if NUMPY_LT_2_0:
            # Ensure ndim>=1 so that comparison is done using the angle dtype.
            # Otherwise, e.g., np.array(np.pi/2, 'f4') > np.pi/2 will yield True.
            angles_view = angles_view[np.newaxis]
    
>       if np.any(np.abs(angles_view) > limit):
E       RuntimeWarning: overflow encountered in cast
```